core/lib/zksync_core/src/api_server/tx_sender/master_pool_sink.rs

```diff
@@ -32,6 +56,11 @@ impl TxSink for MasterPoolSink {
             .await;
 
         APP_METRICS.processed_txs[&TxStage::Mempool(submission_res_handle)].inc();
+        self.inflight_requests
```

Comment:
`connection_tagged().await?` may fail and early exit, in which case the entry won't be cleaned up. Do the cleanup unconditionally, or at least add a todo if in such case the whole server expected to fail anyway

Reply:
Right, also `insert_transaction_l2` can panic, I changed it to error and handle it as well